### ISSUE TITLE
Security system in case of authorities pressure

### ISSUE THREAD

**Issue by regnveig** [oxen-io/session-android#1305]
Dear colleagues,

I'd like to request a feature which would be very useful in some countries, and mine, too.

Unfortunately, sometimes there are situations when you are forced (by tortures or threats) to enter the password of your mobile device. It doesn't matter who forced you--your abusive parents/partner, or law enforcers, or criminals. It is always a risk, and sometimes a grave danger.

So, it would be great to have a system which urgently wipes out the messenger database, or deceive an unauthorized person.

I think the trigger of that system can be put into the password screen.

- password N1: it's all right, open the messenger
- password N2: deceive the unskilled oppressor: open the messenger, but show only contacts without important info (i.e., preset channels, or dialogs changed by the user, or another account), and limited settings.
- password N3: law enforcers mode: wipe the database out and delete all keys.

Or something like that.

Thank you!

**Comment by digitalnb** [comment]
I agree.. When they use tax money to buy private data and the government being the 3rd parties now. This is needed. 
This is just a platform. People don't need to use the platform illegally! Or do harm. It's a shame people like that have to ruin it for others though... 
Anyways.. I agree. People have the right to have privacy.

**Comment by regnveig** [comment]
Thank you for your answer @digitalnb! I agree that it should be an option in the settings. 

I don't agree, though, when you call such precautions paranoia. Thousands of people in my country were tortured by the police, because they speak out against war, corruption, and oppression, or because they are gay. It's simply a fact. Most of them are peaceful, good people, and never were criminals. But they are tortured, and forced to show their private correspondence, and then imprisoned or even killed because of the words they typed to their friends and partners. 

IMHO, freedom of those people has a higher priority than justice for some criminals.

But of course I agree that implementation of such a feature would create some potential difficulties, so it's up to developers.

**Comment by digitalnb** [comment]
Yeah that sounds like madness. 
They do have ordinary password i saw now as protection for private conversations, but you are right that a function like you explained could be good for sure. Just some local command, that removes everything.
Session should improve the crypto though. They should generate their own crypto that's way strong. Or use stronger encryption. The best one out there.

**Comment by dsonozaki** [comment]
@digitalnb, @regnveig , I was thinking about the same thing, and I want to create the fork that will improve Session functionality using some features of "[Partisan telegram](https://github.com/wrwrabbit/Partisan-Telegram-Android)" - the application by cyber partisans of Belarus that add to the Telegram the system of fake passwords. When user enter these passwords application can imperceptibly delete selected chats or log out from selected accounts. However, I am very unexperienced Android developer (just ended updating my first app, and it's still really buggy), and I am not sure I can work with this code alone, so I would be glad if someone watching this thread will help.

**Comment by keybreak** [comment]
First of all - **brilliant**, and super-important idea!
I was thinking in the same way for years.

Few thoughts:
> password N2: deceive an unskilled oppressor: open the messenger, but show only screen without any important info (e.g., some neutral channels, or dialogs previously chosen by the user, or even another account), and limited settings.

I'd like to add that best way here is to just do placeholder dummy dialogs (somewhat randomized, in case your adversary is smart), placeholder dummy fake ID (crucial), placeholder dummy settings.

> password N3: law enforcers mode: wipe the database out and delete all the keys.
+1 for that, BUT i believe that is one crucial detail:
1. Such action should be performed in the background
2. Show exactly same fake stuff as **password N2**, because if you don't - you **can** be forced to give up something real with a hammer to your face. It's super-important in my view to "seem" co-operative in such case, while actually wipe anything real.

__________

Generally speaking though, it's best to do on OS level, like GrapheneOS profiles...in order for all important data to be safe that way, but that's not for 100% of users obviously (sigh...)

I'm also a bit worried of one scenario if your phone has been taken (by thieves or government) **before** you've been able to lock it...That's the most dangerous situation without mitigation possibility so far (you can't logout and nuke existing account from another device now)
